src/Razor/src/Microsoft.NET.Sdk.Razor/GenerateStaticWebAsssetsPropsFile.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class GenerateStaticWebAsssetsPropsFile : Task
+    {
+        private const string SourceType = "SourceType";
+        private const string SourceId = "SourceId";
+        private const string ContentRoot = "ContentRoot";
+        private const string BasePath = "BasePath";
+        private const string RelativePath = "RelativePath";
+
+        [Required]
+        public string TargetPropsFilePath { get; set; }
+
+        [Required]
+        public ITaskItem[] StaticWebAssets { get; set; }
+
+        public override bool Execute()
+        {
+            if (!ValidateArguments())
+            {
+                return false;
+            }
+
+            return ExecuteCore();
+        }
+
+        private bool ExecuteCore()
+        {
+            if (StaticWebAssets.Length == 0)
+            {
+                return !Log.HasLoggedErrors;
+            }
+
+            var template = StaticWebAssets[0];
+
+            var document = new XDocument(new XDeclaration("1.0", "utf-8", "yes"));
+            var root = new XElement(
+                "Project",
+                new XElement("ItemGroup",
+                    new XElement("StaticWebAsset",
+                        new XAttribute("Include", @"$(MSBuildThisFileDirectory)..\staticwebassets\**"),
```

Comment:
Should this list individual files? Avoids MSBuild from evaluation the globbing on every build

Reply:
I'm not sure it'll matter a lot. Can we see how it works in preview and decide on expanding later?